### PR TITLE
Add toggleMenuBarOnAltPressed option to BrowserWindow.

### PR DESCRIPTION
**Pull request by tuvistavie**
Hi,

I recently started using Atom and had some issues with the `Alt` key
toggling menu bar, so I added an option to disable this behavior.

The option can be passed when instanciating a `BrowserWindow`,
or be changed using the `setToggleMenuBarOnAltPressed`method
on a `BrowserWindow` instance.
I also added a `doesToggleMenuBarOnAltPressed` method
to retrieve the current value of the option.
The option is disabled by default, so this will not break or change the behavior of
existing apps.

### Background

In some apps, such as Atom, the menu bar can be restored in other ways,
and as it is not something that is done so often (at least for my usage),
I think this option makes sense.

I will explain my issue in Atom, which is my actual motivation for this option.
I use [XMonad](http://xmonad.org/), and my shortcut to change window focus is `alt+j`.
When I press `alt`, [`menu_bar_alt_pressed` becomes true](https://github.com/atom/electron/blob/master/atom/browser/native_window_views.cc#L864).
Then, when I press `j`, the keydown event is intercepted by XMonad,
and the focus changes to some other app, let's say my browser.
As Atom did not receive the `j` keydown event, `menu_bar_alt_pressed` is still true.
When I then want to go back go Atom, I press `alt+j` again. I usually release the
`j` button instantly, and the `alt` after, so the `j` keyup is not handled by atom, but
the `alt` keyup event is, which makes [the condition for toggling the menu bar](https://github.com/atom/electron/blob/master/atom/browser/native_window_views.cc#L867) true.
So basically each time I change focus to some other app and go back to Atom,
the menu bar is toggled, which is a little annoying.

### CODE REVIEW

**Comment by zcbenz** [comment]
Why not just disables the auto-hide menu?

**Comment by tuvistavie** [comment]
@zcbenz I want the menu to be always hidden, not always shown.

**Comment by zcbenz** [comment]
@tuvistavie There is `setMenuBarVisibility` API to show/hide the menu.

**Comment by tuvistavie** [comment]
@zcbenz Great, that's perfect! Thank you.

**Comment by theFroh** [comment]
Does using `setMenuBarVisibility` to disable the menu appearing prevent the menu taking focus as well?

**Comment by tuvistavie** [comment]
@theFroh It does not seem to. If I press `alt+e` (at least in Atom), the edit menu and the menu bar shows up even with `setMenuBarVisibility` set to `false`. However, if the shortcut is assigned, the assigned binding takes advantage.
